### PR TITLE
[prometheus-node-exporter] Update labels to match best practices

### PR DESCRIPTION
**Pull request by QuentinBisson**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

This PR updates the node exporter labels to match the best practice described here: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
This is a breaking change for users which already have installed an older version.
So upgradepath has to be defined and documented.

**Comment by QuentinBisson** [comment]
@monotek do we have an example I could use as I think it's not the first time this happened in this repo ? :D

**Comment by QuentinBisson** [comment]
I found an example for the stackdriver exporter that exactly matches what I needed :)

**Comment by QuentinBisson** [comment]
@monotek what do you think about this one ?

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Please try the upgrade out at least once and fix the readme regarding to the errors you'll find.

**Comment by QuentinBisson** [comment]
@monotek I will try the upgrade but the readme was changed https://github.com/prometheus-community/helm-charts/pull/2212/commits/af533d6fd5765b143dcbd70dc50667dfdfe80b23

**Review by zanhsieh — DISMISSED** [review]

**Review by mrueg — DISMISSED** [review]

**Comment by mrueg** [comment]
@monotek looks like I can't see your review in this one. Anything that it still open from your side here?

**Comment by monotek** [comment]
Yes, my posted suggestion. Not the deployment but the deaminset has to be deleted.

File: charts/prometheus-node-exporter/README.md [hunk 1]
```diff
@@ -41,6 +41,17 @@ helm upgrade [RELEASE_NAME] [CHART] --install
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
+### 3.x to 4.x
+
+Starting from version 4.0.0, the `node exporter` chart is using the [Kubernetes recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). Therefore you have to delete the deployment before you upgrade.
+
+```console
+kubectl delete daemonset -l app=prometheus-node-exporter
```
Comment (mrueg) [inline charts/prometheus-node-exporter/README.md:49]: @monotek looks like the suggestion here is to delete the daemonset now, am I missing anything?

**Review by monotek — CHANGES_REQUESTED** [review]
Please rebase and fix readme.

**Comment by QuentinBisson** [comment]
@monotek the readme is fixed

File: charts/prometheus-node-exporter/README.md [hunk 2]
```diff
@@ -41,6 +41,17 @@ helm upgrade [RELEASE_NAME] [CHART] --install
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
+### 3.x to 4.x
+
+Starting from version 4.0.0, the `node exporter` chart is using the [Kubernetes recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). Therefore you have to delete the deployment before you upgrade.
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply charts/prometheus-node-exporter/README.md]: The command is just below :)

File: charts/prometheus-node-exporter/README.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
-# Prometheus Node Exporter
+# Prometheus `Node Exporter`
 
 Prometheus exporter for hardware and OS metrics exposed by *NIX kernels, written in Go with pluggable metric collectors.
 
-This chart bootstraps a prometheus [Node Exporter](http://github.com/prometheus/node_exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+This chart bootstraps a prometheus [`Node Exporter`](http://github.com/prometheus/node_exporter) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
```
Comment (monotek) [inline charts/prometheus-node-exporter/README.md:5]:
```suggestion
This chart bootstraps a prometheus [`Node Exporter`](http://github.com/prometheus/node_exporter) daemonset on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Servicemonitor Labels need to be changed too

**Comment by QuentinBisson** [comment]
I signed off your commit but this should be fine now, sorry about all those missed parts :(

**Review by monotek — APPROVED** [review]

**Pull request merged**